windows/src/engine/kmcomapi/processes/keyboard/Keyman.System.Process.KPUninstallKeyboardLanguage.pas

```diff
@@ -0,0 +1,382 @@
+unit Keyman.System.Process.KPUninstallKeyboardLanguage;
+
+interface
+
+uses
+  System.Win.Registry,
+  Winapi.Windows,
+
+  kpbase,
+  KeymanContext,
+  Windows8LanguageList,
+  Winapi.msctf,
+  keyman_msctf;
+
+type
+  TKPUninstallKeyboardLanguage = class(TKPBase)
+    /// <summary>Unregisters all TIPs from the Local Machine registry for a given
+    /// keyboard, including transient profiles</summary>
+    /// <param name="KeyboardID">The ID of the keyboard</param>
+    procedure UnregisterTip(const KeyboardID: string); overload;
+
+    /// <summary>Uninstalls a TIP for the current user for a given keyboard</summary>
+    /// <param name="KeyboardID">The ID of the keyboard</param>
+    /// <param name="BCP47Tag">This should be a BCP47 tag that is currently installed</param>
+    procedure UninstallTip(const KeyboardID, BCP47Tag: string); overload;
+
+    /// <summary>Uninstalls a TIP for the current user for a given keyboard</summary>
+    /// <param name="KeyboardID">The ID of the keyboard</param>
+    /// <param name="LangID">The transient profile to uninstall, $2000, $2400, $2800, $2C00</param>
+    procedure UninstallTip(const KeyboardID: string; LangID: Word); overload;
+
+    /// <summary>Uninstalls all TIPs for the current user for a given keyboard</summary>
+    /// <param name="KeyboardID">The ID of the keyboard</param>
+    procedure UninstallTip(const KeyboardID: string); overload;
+
+  protected
+    /// <summary>Unregisters a TIP from the Local Machine registry</summary>
+    /// <param name="BCP47Tag">This should be a BCP47 tag</param>
+    /// <param name="KeyboardID">The ID of the keyboard</param>
```

Comment:
Does it matter in the Delphi IDE if the parameters are listed in reverse order?

Reply:
It won't matter but I should fix it for clarity!